packages/documentation/docs/controls/_number-field_styleguide.mdx

```diff
@@ -0,0 +1,51 @@
+The Number input component allows users to enter and adjust numerical values.
+
+
+![Number input overview](https://www.figma.com/design/wEptRgAezDU1z80Cn3eZ0o/iX-Pattern-Illustrations?node-id=3805-24565&t=DtCmoFcLwhf7ke3S-4)
+
+1. Label
+2. Required field indicator (optional)
+3. Value
+4. Stepper buttons (optional)
+5. Input field
+6. Helper or feedback text
+
+## Options
+
+- **Label**: [See form field](./forms/forms-field.md)
+- **Value**: [See form field](./forms/forms-field.md)
+- **Required**: [See form field](./forms/forms-field.md)
+- **Helper text**: [See form field](./forms/forms-field.md)
+- **Feedback text**: [See form field](./forms/forms-field.md)
+- **Show text as tooltip**: [See form field](./forms/forms-field.md)
+- **Placeholder text**: [See form field](./forms/forms-field.md)
+- **Allowed characters pattern**: Define which characters are allowed to be entered. Typically, we use this to reject excluded characters like decimal points with a shaking animation already on value change while typing.
```

Comment:
Specify the allowed characters for input. We typically use this to reject invalid characters, such as decimal points, by triggering a shaking animation immediately upon value change during typing.